### PR TITLE
masurca 3.2.2

### PR DESCRIPTION
**Pull request by philippbayer**
Satisfy Rubocop

### Have you:

- [X] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [X] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [ ] Written a sensible test? (The best test is to compile and run a sample program.)

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [X] Not altered the `bottle` section?
- [X] Checked that the tests still pass?

Update is recommended by the author on his blog.

I was able to delete some old workarounds and had to add a new one. Pretty sure it still won't work on Mac OS as a few lines still contain references to Mac's missing `mremap`.

Still getting this error on `brew audit`, unsure how to fix:

<pre> 
   * Non-executables were installed to "/gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin"
    The offending files are:
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/MasurcaSoap.pm
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/MasurcaConf.pm
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/MasurcaCelera.pm
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/run_ECR.sh
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/MasurcaCommon.pm
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/MasurcaPaths.pm
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/MasurcaSuperReads.pm
      /gpfs1/homes/pbayer/.linuxbrew/opt/masurca/bin/fix_unitigs.sh
</pre>

### CODE REVIEW

**Comment by tseemann** [comment]
Wow, this formula is quite a complicated one.

Some comments:

* do you need to add `samtools`, `mummer` etc to the `depends_on` ? or is it using the library interface instead of the command line tools?
* there is a `depends_on` Linux clause but there is also `if Mac.OS?` conditionals
* `.pm` files are Perl modules and should not be in a `bin` folder. Resolving this is a bit tricky; they need to be moved to a different folder, then the perl script using them needs to be updated to add a `use lib 'new_dir`;` line, or `PERL5LIB` set custom in a wrapper script

**Comment by philippbayer** [comment]
Yes it looks like it has grown quite a bit in the last years! Thanks for the review.


    does it have precompiled @linux binaries for Celera Assembler embedded in the CA/*/bin folder?

Before compilation there are three Perl scripts in those two folders (CA/bin, CA8/bin), no precompiled binaries.

    do you need to add samtools, mummer, soapdenovo etc to the depends_on ? or is it using the library interface instead of the command line tools?

Oof good question. For mummer, it uses the library interface in bin/refine_alignments.pl (`mummer::align_sequences`).  For soapdenovo it runs the command line tools internally (which is optional, Celera Assembler is standard way). For samtools, the included Celera Assembler (CA) needs the ('old' before htslib) samtools headers,  I can't find code that runs the command line samtools.

    there is a depends_on Linux clause but there is also if Mac.OS? conditionals

Yes, a past version seems to have broken the Mac.OS support so we have some overhang. I guess it's better to leave the depends_on clause and delete everything else regarding Mac, will do that

    .pm files are Perl modules and should not be in a bin folder. Resolving this is a bit tricky; they need to be moved to a different folder, then the perl script using them needs to be updated to add a use lib 'new_dir'; line, or PERL5LIB set custom in a wrapper script

Is there any precedent for this in any of the other recipes? quick skim reveals nothing

    it says ENV.deparallelize but the provided install.sh explicity sets -j CPUS in it

Aha! Some more overhang, deleting that line and reinstalling masurca seems to work fine.

**Comment by sjackman** [comment]
> .pm files are Perl modules and should not be in a bin folder.

Another option is to install everything in `libexec` rather than `prefix`, and then symlink the actual executables from `#{libexec}/bin/foo` to `#{bin}/foo`.

File: masurca.rb [hunk 1]
```diff
@@ -55,11 +44,23 @@ def install
...
+    rm bin/"exact-tandems"
+    rm bin/"mummer"
+    rm bin/"mummerplot"
+    rm bin/"nucmer"
+    rm bin/"promer"
+    rm bin/"repeat-match"
+    rm bin/"show-coords"
+    rm bin/"show-diff"
+    rm bin/"show-snps"
+    rm bin/"show-tiling"
```
Comment (sjackman) [inline masurca.rb]: I suggest…
```ruby
cd bin { rm %w[combineMUMs delta-filter … show-snps show-tiling] }
```

File: masurca.rb [hunk 2]
```diff
@@ -34,15 +32,6 @@ class Masurca < Formula
...
-    if OS.mac?
-      # Fix cp: CA/Linux-amd64/bin/*: No such file or directory
-      inreplace "install.sh", "Linux-amd64", "Darwin-amd64"
-    elsif OS.linux?
-      # Fix libstdc++.so: undefined reference to `clock_gettime@GLIBC_2.17'
-      inreplace "CA/src/c_make.as", %r{ARCH_LIB *= /usr/lib64 /usr/X11R6/lib64}, ""
-    end
-
-    ENV.deparallelize
     ENV["DEST"] = prefix
```
Comment (sjackman) [inline masurca.rb]: Try
```ruby
ENV["DEST"] = libexec
```
and then symlink the primary executables into `bin`.

**Comment by sjackman** [comment]
> For samtools, the included Celera Assembler (CA) needs the ('old' before htslib) samtools headers, I can't find code that runs the command line samtools.

There's a keg-only formula for `samtools@0.1`

**Comment by sjackman** [comment]
```
configure: cannot find Boost headers version >= 1.46.0
```
https://travis-ci.org/Homebrew/homebrew-science/builds/239846076#L1332

@tmozgach Please adopt this PR and update MaSuRCA to 3.2.2. Please add `depends_on "boost"`.

**Comment by philippbayer** [comment]
Sorry stalebot! I totally forgot about this PR, will have a look this week. There's also another update to 3.2.3 to implement!

**Comment by sjackman** [comment]
```
checking for Boost headers version >= 1.46.0... no
configure: cannot find Boost headers version >= 1.46.0
```
https://circleci.com/gh/Homebrew/homebrew-science/5074
Should `depends_on "boost"` or possible `depends_on "boost" => :build` depending on whether they use the libraries or only the headers.

**Comment by sjackman** [comment]
@philippbayer Are you interested in updating this PR?

**Comment by sjackman** [comment]
Superseded by `masurca` 3.2.3 PR https://github.com/Homebrew/homebrew-science/pull/6497

Note that this tap Homebrew/science is being retired at the end of the year. The next revision of `masurca` can migrate to https://github.com/Linuxbrew/homebrew-bio/

**Comment by philippbayer** [comment]
:+1: Thanks for not doing more with this, life really took over the last few months

**Pull request closed without merging**